hybris/tests/Makefile.am

```diff
@@ -16,7 +16,18 @@ bin_PROGRAMS = \
 	test_wifi \
 	test_hwc2 \
 	test_hwcomposer \
-	test_nfc
+	test_nfc \
+	test_dlopen
+
+
+test_dlopen_SOURCES = test_dlopen.c
+test_dlopen_CFLAGS = \
+	-I$(top_srcdir)/include \
+	-I$(top_srcdir)/hybris/include \
+	-I$(top_srcdir)/include/android
+test_dlopen_LDADD = \
+	$(top_builddir)/common/libhybris-common.la \
+	$(top_builddir)/hardware/libhardware.la
 
 test_audio_SOURCES = test_audio.c
 test_audio_CFLAGS = \
```

Comment:
Adding C99 here was needed for me to fix building

Reply:
I also have no issue.
Could you please show the error message ?

Reply:
Without the -std=c99 flag c89 is used here which doesn't  allow the initialization of variables inside the loop

https://stackoverflow.com/questions/1287863/c-for-loop-int-initial-declaration

Reply:
Well, even I have add the c89 manually , I can still build it successfully.
gcc -DHAVE_CONFIG_H -I. -I..   -I/home/android_head/ -I../include -I../hybris/include -I../include/android -g -O2 **-std=c89** -MT test_dlopen-test_dlopen.o -MD -MP -MF .deps/test_dlopen-test_dlopen.Tpo -c -o test_dlopen-test_dlopen.o `test -f 'test_dlopen.c' || echo './'`test_dlopen.c

Maybe I need to downgrade my GCC? Since I can't reproduce your two issues at all.

Reply:
The comment was about the audio test not the dlopen test.

Reply:
The audio header from android q includes a loop that uses c99 style loops.